test/msw-api/setup-server/scenarios/concurrency/process-one.test.ts

```diff
@@ -0,0 +1,89 @@
+import { rest, graphql } from 'msw'
+import { setupServer } from 'msw/node'
+import { makeRequest } from './api'
+import {
+  sharedServer,
+  sleep,
+  endpoint,
+  getNow,
+  beforeAllResolve,
+  beforeAllPromise,
+  sleepDelay,
+} from './shared-server'
+
+describe('concurrency test', () => {
+  beforeAll(async () => {
+    sharedServer.listen()
+    beforeAllResolve()
+  })
+
+  afterAll(async () => {
+    sharedServer.close()
+  })
+
+  it.concurrent('process one tests', async () => {
+    await beforeAllPromise
+
+    const mockFn = jest.fn()
+
+    console.log('step 1 - process one attach handler to server: '),
+      sharedServer.use(
+        rest.get(`${endpoint}/:processId`, (req, res, ctx) => {
+          console.log(
+            'step 4 - process one handler resolver triggered: ',
+            getNow(),
+          )
+
+          mockFn(req.params.processId)
+
+          return res(
+            ctx.json({
+              text: 'process one handler resolver response',
+            }),
+          )
+        }),
+      )
+
+    const handlers = sharedServer.listHandlers()
+
+    await sleep(sleepDelay)
+
+    console.log('step 3 - process one make request: ')
+
+    await makeRequest(`${endpoint}/one`)
+
+    expect(mockFn).toHaveBeenCalledWith('one')
+
+    console.log('process one end: ', handlers)
+  })
+
+  it.concurrent('process two tests', async () => {
+    await beforeAllPromise
+
+    const mockFn = jest.fn()
+
+    console.log('step 2 - process two attach handler to server: '),
+      sharedServer.use(
+        rest.get(`${endpoint}/:processId`, (req, res, ctx) => {
+          console.log(
```

Comment:
I think we should turn these logs from purely observational to the data we actually assert in the test. We're doing a similar pattern with the life-cycle events, you can take a look at it here:

https://github.com/mswjs/msw/blob/b93aca9f165294d064517990e6c55aee7557d9b1/test/msw-api/setup-server/life-cycle-events/on.test.ts#L94

Checking that the mock function was called with the correct arguments still doesn't guarantee that the entire request processing pipeline was correct.